### PR TITLE
Optimization in module `sympy/holonomic` and Implementing RK4 for numerical computation.

### PR DESCRIPTION
**Pull request by shubhamtibra**
- [x] Add Runge Kutta 4th order
- [x] Add tests

### CODE REVIEW

**Comment by shubhamtibra** [comment]
I have added some tests. Ping @certik @jksuom .

**Comment by shubhamtibra** [comment]
I have added the RK4 method and it is used as default in `.evalf()`. Ping @certik @jksuom . Let me know if this is Okay.

**Comment by certik** [comment]
Perfect, this looks very good. The rk4 is much more accurate for the same
mesh, as expected.

+1 to merge from me.

**Comment by jksuom** [comment]
Looks very good to me. Merging.

File: sympy/holonomic/holonomic.py [hunk 1]
```diff
@@ -1036,14 +1036,24 @@ def evalf(self, points, **kwargs):
         >>> # a straight line on the real axis from (0 to 1)
         >>> r = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1]
 
-        >>> HolonomicFunction(Dx - 1, x, 0, [1]).evalf(r)  # e^1
+        # using Runge-Kutta 4th order on e^x from 0.1 to 1.
+        # exact solution at 1 is 2.71828182845905
+        >>> HolonomicFunction(Dx - 1, x, 0, [1]).evalf(r)
+        [1.10517083333333, 1.22140257085069, 1.34985849706254, 1.49182424008069,
```
Comment (asmeurer) [inline sympy/holonomic/holonomic.py]: This line has trailing whitespace, and fails the code quality tests. I'm curious how this PR got merged with code quality tests failing.

Comment (shubhamtibra) [inline sympy/holonomic/holonomic.py]: All the checks had passed when this was merged. Does this need to be resolved right now?

Comment (asmeurer) [inline sympy/holonomic/holonomic.py]: I fixed it at https://github.com/sympy/sympy/pull/11205. See also https://github.com/sympy/sympy/issues/11204.

Comment (asmeurer) [inline sympy/holonomic/holonomic.py]: For now, I would recommend configuring your editor to highlight trailing whitespace, and, if possible, to clear it when you save.

**Pull request merged**